bin/generate_tests.py

```diff
@@ -101,6 +101,47 @@ def parse_datetime(string, strip_module=False):
         return result.replace("datetime.", "", 1)
     return result
 
+INVALID_ESCAPE_RE = re.compile(
+    r"""
+    \\(?!                           # a backslash NOT followed by
+        newline                     # the literal newline
+    |[                              # OR precisely one of
+        \\                          # another backslash
+        '                           # the single quote
+        "                           # the double quote
+        a                           # the ASCII bell
+        b                           # the ASCII backspace
+        f                           # the ASCII formfeed
+        n                           # the ASCII linefeed
+        r                           # the ASCII carriage return
+        t                           # the ASCII horizontal tab
+        v                           # the ASCII vertical tab
+    ]|                              # OR
+        o(?:[0-8]{1,3})             # an octal value
+    |                               # OR
+        x(?:[0-9A-Fa-f]{2})         # a hexidecimal value
+    |                               # OR
+        N                           # a unicode char name composed of
+        \{                          # an opening brace
+            [A-Z][A-Z\ \-]*[A-Z]    # uppercase WORD, WORDs (or WORD-WORDs)
+        \}                          # and a closing brace
+    |                               # OR
+        u(?:[0-9A-Fa-f]{4})         # a 16-bit unicode char
+    |                               # OR
+        U(?:[0-9A-Fa-f]{8})         # a 32-bit unicode char
+    )""", flags=re.VERBOSE)
```

Comment:
I had no idea Python `re` allowed you to expand out a regex pattern like this with line comments and everything.

Reply:
Yep ... good ol `re.VERBOSE`.